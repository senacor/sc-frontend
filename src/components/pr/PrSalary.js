import React from 'react';

import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import moment from 'moment';
import { withStyles } from '@material-ui/core/styles/index';

const styles = theme => ({
  tableRow: {
    backgroundColor: theme.palette.primary['50'],
    width: '100%'
  },
  table: {
    tableLayout: 'fixed',
    width: '100%',
    marginBottom: '1px'
  }
});

class PrSalary extends React.Component {
  render() {
    const { prDetail, classes } = this.props;

    return (
      <Paper>
        <Table className={classes.table}>
          <TableHead>
            <TableRow>
              <TableCell colSpan={2}>Gehalt seit:</TableCell>
              <TableCell colSpan={2}>OTE:</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {prDetail.employee.salaries ? (
              prDetail.employee.salaries
                .sort((a, b) => a.validFrom < b.validFrom)
                .map(value => {
                  return (
                    <TableRow
                      key={prDetail.employee.salaries.indexOf(value)}
                      className={
                        prDetail.employee.salaries.indexOf(value) % 2 === 0
                          ? classes.tableRow
                          : ''
                      }
                    >
                      <TableCell colSpan={2}>
                        {moment(value.validFrom).format('DD.MM.YY')}
                      </TableCell>

                      <TableCell colSpan={2}>{value.ote}</TableCell>
                    </TableRow>
                  );
                })
            ) : (
              <TableRow />
            )}
          </TableBody>
        </Table>
      </Paper>
    );
  }
}
export default withStyles(styles)(PrSalary);
