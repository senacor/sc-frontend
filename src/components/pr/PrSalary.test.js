import React from 'react';
import PrSalary from './PrSalary';
import { createShallow } from 'material-ui/test-utils';

describe('PrSalary Component', () => {
  let shallow = createShallow({ dive: true });

  const prById = {
    id: 1,
    employee: {
      id: 1,
      login: 'lschäfer',
      firstName: 'Lionel',
      lastName: 'Schäfer',
      title: 'DR',
      email: 'lionel.schäfer@senacor.com',
      entryDate: '2004-05-10',
      costCenters: [
        {
          name: 'WRJ',
          supervisorId: 1,
          costCenterId: 10
        },
        {
          name: 'KLC',
          supervisorId: 1,
          costCenterId: 50
        },
        {
          name: 'NPU',
          supervisorId: 1,
          costCenterId: 30
        },
        {
          name: 'CYP',
          supervisorId: 1,
          costCenterId: 40
        },
        {
          name: 'BOH',
          supervisorId: 1,
          costCenterId: 20
        }
      ],
      salaries: [
        {
          workingHoursPerWeek: 40,
          ote: 45000,
          fte: 1,
          basicSalary: 40500,
          validFrom: '2004-05-10'
        },
        {
          workingHoursPerWeek: 40,
          ote: 50000,
          fte: 1,
          basicSalary: 40000,
          validFrom: '2005-05-10'
        }
      ],
      employment: {
        endOfProbationPeriod: '2004-05-10',
        jobGradings: [
          {
            positionId: 1,
            salaryLevel: 1,
            validFrom: '2004-05-09',
            competence: 'DEVELOPMENT'
          },
          {
            positionId: 2,
            salaryLevel: 2,
            validFrom: '2005-05-09',
            competence: 'DEVELOPMENT'
          }
        ],
        leaves: {
          unpaidLeave: [
            {
              from: '2004-05-10',
              to: '2005-03-31'
            }
          ],
          maternityLeave: [
            {
              from: '2004-05-10',
              to: '2004-10-11'
            }
          ],
          parentalLeave: [
            {
              from: '2004-05-10',
              to: '2004-08-28'
            }
          ],
          sabbatical: [
            {
              from: '2004-05-10',
              to: '2004-06-07'
            }
          ]
        }
      },
      staffings: [
        {
          costCenterId: 34,
          date: '2018-03-16'
        },
        {
          costCenterId: 8,
          date: '2018-04-04'
        }
      ],
      scorecards: []
    },
    supervisor: 'ttran',
    occasion: 'ON_DEMAND',
    status: 'PREPARATION',
    deadline: '2015-05-11',
    _links: {
      self: {
        href: 'http://localhost:8010/api/v1/prs/1'
      }
    }
  };

  it('tests the addAllValues function', () => {
    const output = [
      { validFrom: '2004-05-10', reason: 'Ende der Probezeit' },
      { validFrom: '2004-05-10', ote: 45000, fte: 1 },
      { validFrom: '2005-05-10', ote: 50000, fte: 1 },
      {
        validFrom: '2004-05-10',
        validTo: '2005-03-31',
        reason: 'Unbezahlter Urlaub'
      },
      {
        validFrom: '2004-05-10',
        validTo: '2004-10-11',
        reason: 'Mutterschutz'
      },
      { validFrom: '2004-05-10', validTo: '2004-08-28', reason: 'Elternzeit' },
      {
        validFrom: '2004-05-10',
        validTo: '2004-06-07',
        reason: 'Forschungsurlaub'
      }
    ];
    const component = shallow(<PrSalary prById={prById} />);
    expect(component.instance().addAllValues(prById)).toEqual(output);
  });

  it('displays the PrSalary', () => {
    const component = shallow(<PrSalary prById={prById} />);
    expect(component).toMatchSnapshot();
  });
});
