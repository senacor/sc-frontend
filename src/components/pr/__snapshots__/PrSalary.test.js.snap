// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrSalary Component displays the PrSalary 1`] = `
<WithStyles(Paper)>
  <WithStyles(Table)
    className="PrSalary-table-2"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Gehalt seit:
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          OTE:
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={true}
        >
          FTE:
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        className="PrSalary-tableRow-1"
        key="0"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.05
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          50000
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={true}
        >
          1
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className=""
        key="1"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04 - 31.03.05
           
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Unbezahlter Urlaub
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className="PrSalary-tableRow-1"
        key="2"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04 - 11.10.04
           
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Mutterschutz
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className=""
        key="3"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04 - 28.08.04
           
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Elternzeit
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className="PrSalary-tableRow-1"
        key="4"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04 - 07.06.04
           
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Forschungsurlaub
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className=""
        key="5"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          Ende der Probezeit
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        className="PrSalary-tableRow-1"
        key="6"
      >
        <WithStyles(TableCell)
          colSpan={2}
        >
          10.05.04
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          colSpan={2}
        >
          45000
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          numeric={true}
        >
          1
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Paper)>
`;
